desc/objectives/_geometry.py

```diff
@@ -357,3 +363,147 @@ def compute(self, *args, **kwargs):
+            name=name,
+        )
+
+    def build(self, eq, use_jit=True, verbose=1):
+        """Build constant arrays.
+
+        Parameters
+        ----------
+        eq : Equilibrium, optional
+            Equilibrium that will be optimized to satisfy the Objective.
+        use_jit : bool, optional
+            Whether to just-in-time compile the objective and derivatives.
+        verbose : int, optional
+            Level of output.
+
+        """
+        if self._surface_grid is None:
+            self._surface_grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
+        if self._plasma_grid is None:
+            self._plasma_grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP)
+        if not np.allclose(self._surface_grid.nodes[:, 0], 1):
+            warnings.warn("Surface grid includes off-surface pts, should be rho=1")
+        if not np.allclose(self._plasma_grid.nodes[:, 0], 1):
+            warnings.warn("Plasma grid includes interior points, should be rho=1")
+
+        self._dim_f = self._plasma_grid.num_nodes
+        self._data_keys = ["R", "phi", "Z"]
+        self._args = get_params(self._data_keys)
+
+        timer = Timer()
+        if verbose > 0:
+            print("Precomputing transforms")
+        timer.start("Precomputing transforms")
+
+        self._surface_coords = self._surface.compute_coordinates(
+            grid=self._surface_grid, basis="xyz"
+        )
+        self._profiles = get_profiles(self._data_keys, eq=eq, grid=self._plasma_grid)
+        self._transforms = get_transforms(
+            self._data_keys, eq=eq, grid=self._plasma_grid
+        )
+
+        timer.stop("Precomputing transforms")
+        if verbose > 1:
+            timer.disp("Precomputing transforms")
+
+        if self._normalize:
+            scales = compute_scaling_factors(eq)
+            self._normalization = scales["a"]
+
+        super().build(eq=eq, use_jit=use_jit, verbose=verbose)
+
+    def compute(self, *args, **kwargs):
+        """Compute minimum plasma-surface distance.
+
+        Parameters
+        ----------
+        R_lmn : ndarray
+            Spectral coefficients of R(rho,theta,zeta) -- flux surface R coordinate (m).
+        Z_lmn : ndarray
+            Spectral coefficients of Z(rho,theta,zeta) -- flux surface Z coordinate (m).
+
+        Returns
+        -------
+        d : ndarray, shape(plasma_grid.num_nodes,)
+            For each point in the plasma grid, approximate minimum distance to bounding
+            surface.
+
+        """
+        params = self._parse_args(*args, **kwargs)
+        data = compute_fun(
+            self._data_keys,
+            params=params,
+            transforms=self._transforms,
+            profiles=self._profiles,
+        )
+        plasma_coords = rpz2xyz(jnp.array([data["R"], data["phi"], data["Z"]]).T)
+        d = jnp.linalg.norm(
+            plasma_coords[:, None, :] - self._surface_coords[None, :, :], axis=-1
```

Comment:
And does this only check distances at the given phi cross-sections from grid zeta values, or does it check distances btwn every point on the surface against every other point (so even phi=0 and phi=pi values are compared), I assume not the latter?

Reply:
Yes, this is the distance matrix between every point on the plasma boundary and every point on the bounding surface, and it will look at every cross section. If the grids are different resolution or staggered or something then the closest point to phi=0 might be at phi=2pi etc.